core/src/query.rs

```diff
@@ -212,10 +212,11 @@ impl MongoQuery {
         query: &str,
         type_mode: TypeMode,
         max_string_length: Option<u16>,
-    ) -> Result<Self> {
+    ) -> Result<(Self, Diagnostics)> {
+        log::debug!("Preparing query with metadata - query: {query}");
```

Comment:
Adding this here as it's possible for the all to fail in translate and the query would help us.